tests/client.rs

```diff
@@ -395,3 +395,18 @@ async fn echo(req: hyper::Request<Body>) -> Result<hyper::Response<Body>, hyper:
         _ => Ok(hyper::Response::new(req.into_body())),
     }
 }
+
+#[test]
+fn compile_tests() {
+    #[allow(dead_code)]
+    async fn post_ping<'buf>(port: u16, rx_buf: &'buf mut [u8]) -> &'buf mut [u8] {
```

Comment:
Maybe rename to "buffer_lifetime" or something like it

Reply:
Sorry for pasting your own repro in verbatim :P